modules/gateway/gateway.go

```diff
@@ -2,63 +2,90 @@ package gateway
 
 import (
 	"errors"
+	"net"
 	"sync"
 
 	"github.com/NebulousLabs/Sia/consensus"
 	"github.com/NebulousLabs/Sia/modules"
-	"github.com/NebulousLabs/Sia/network"
+)
+
+const (
+	// maxStrikes is the number of "strikes" that can be incurred by a peer
+	// before it will be removed.
+	// TODO: need a way to whitelist peers (e.g. hosts)
+	maxStrikes = 5
 )
 
 var (
+	// TODO: unexport these, or move them to modules
```

Comment:
I'm not sure which to do.

If we move them to modules, we'll be obligated to maintain them, which means we have little flexibility for adjusting the errors. At the same time, if we don't export them, then external forces have no way to check the error.

I think we should move the errors to modules if we're expecting other programs to have different behavior depending on which error gets returned. Otherwise I'd just unexport them.